usr/src/uts/aarch64/promif/prom_utils.c

```diff
@@ -72,6 +73,34 @@ prom_get_prop_int(pnode_t node, const char *name, int def)
 	return value;
 }
 
+uint64_t
+prom_get_prop_u64(pnode_t node, const char *name, uint64_t def)
+{
+	uint64_t prop;
+	uint64_t value = def;
+
+	if (node > 0 && prom_getproplen(node, name) == sizeof(uint64_t)) {
+		prom_getprop(node, name, (caddr_t)&prop);
+		value = ntohll(prop);
+	}
+
+	return value;
+}
+
+uint32_t
+prom_get_prop_u32(pnode_t node, const char *name, uint32_t def)
+{
+	uint32_t prop;
+	uint32_t value = def;
+
+	if (node > 0 && prom_getproplen(node, name) == sizeof(uint32_t)) {
+		prom_getprop(node, name, (caddr_t)&prop);
+		value = ntohl(prop);
+	}
+
+	return value;
```

Comment:
cstyle is going to nag about the unparenthesized returns

Reply:
This file is cstyle hell - I'll do a separate PR to clean the whole thing up.